soroban-sdk/src/auth.rs

```diff
@@ -0,0 +1,15 @@
+use crate::{contracttype, Address, BytesN, RawVal, Symbol, Vec};
+
+#[contracttype(crate_path = "crate", export = false)]
+#[derive(Clone)]
+pub struct AuthContext {
+    pub contract: BytesN<32>,
+    pub fn_name: Symbol,
+    pub args: Vec<RawVal>,
+}
+
+impl AuthContext {
```

Comment:
I've actually tried to expand 'auth' when reasonable. `check_auth` is an exception because the full version is `check_authentication_and_authorization`. I don't feel too strong here though.